### ISSUE TITLE
Return if request was cached or not

### ISSUE THREAD

**Issue by Zyles** [rtheunissen/guzzle-cache-handler#4]
Small feature request...

I need a way to check if the request was fetched from cache or not. Is there a way to do this?

Basically I need to end up with a variable that says $cached = true/false

Thank you.

**Comment by rtheunissen** [comment]
You could check if the body is an instance of `CachedStream`.

``` php
$body = $response->getBody();
$cached = is_a($body, CachedStream::class);
```

**Comment by Zyles** [comment]
Always returns false.

**Comment by rtheunissen** [comment]
Make sure that you're importing the namespace.

``` php
is_a($body, \Concat\Http\Handler\CachedStream::class);
```

or

``` php
use Concat\Http\Handler\CachedStream;

is_a($body, CachedStream::class);
```

**Comment by rtheunissen** [comment]
Do you want to know if the response was made and cached during the last request, or whether the response was fetched from the cache?

**Comment by rtheunissen** [comment]
> if the request was fetched from cache or not

I don't think the `is_a` solution will work for that, as it would also be true for the case where the request was made, and cached. 

I'll come up with a solution, something like `$handler->lastRequestCached()`.

**Comment by Zyles** [comment]
If the result is already cached it should return true, if it is not return false.

Using the namespace it returns true always.

Also it would be prefered to only cache valid responses and avoid 404, 503 status codes etc.

For example you can query http://httpbin.org/status/404 and it will create a cache as you can see. Or 500 error, what if the service is temporarily down? Then up again, it would cache the bad response.

**Comment by rtheunissen** [comment]
Fixed in 9998c809c7084b8dc6c3a8ee2214c3b4963bb05c
